File: 프로그래머스/Level2/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int sum = 0;
    int small = 0, big = people.size() - 1;
        
    sort(people.begin(), people.end(),less<int>());
    
    while(small <= big){
        if(people[small] + people[big] > limit){
            big --;
            answer ++;
        }
        else{
            small ++;
            big --;
            answer ++;
        }
    }
    
    return answer;
}

File: 프로그래머스/Level2/구명보트.js
function solution(people, limit) {
    people = people.sort((a,b) => b-a);
    
    let k = people.length-1, answer = 0;
    
    for(let i=0; i<=k; i++){
        if(people[i] + people[k] > limit)
            answer ++;
        else{
            answer ++;
            k--;
        }
    }
    return answer;
}
